doc/user_guide.md

```diff
@@ -0,0 +1,435 @@
+
+## OFT API
+
+If you are a software developer planning to integrate OFT into one of your programs or scripts, you will probably want to use the OFT API.
+
+Below you find a few short examples of how to use the OFT API. For details check the JavaDoc documentation of the interfaces in the source code.
+
+* [org.itsallcode.openfasttrace.Converter](../src/main/java/org/itsallcoded/openfasttrace/Converter.java)
+* [org.itsallcode.openfasttrace.Reporter](../src/main/java/org/itsallcoded/openfasttrace/Reporter.java)
+
+### Using OFT From Java
+
+The Java interface uses the "fluent programming" paradigm to make the code more compact and easy to read.
+
+#### Using the Converter From Java
+
+The following example code configures a `Converter` to read input form the relative paths `doc`, `src/main/java` and `src/main/test` and output the result to `/tmp/out.xml` in SpecObject format using Unix newlines. 
+
+```JAVA
+import org.itsallcode.openfasttrace.Converter;
+import org.itsallcode.openfasttrace.core.Newline;
+import org.itsallcode.openfasttrace.mode.ConvertMode;
+
+final List<String> inputs = Arrays.asList("doc", "src/main/java", "src/main/test");
+final Converter converter = new ConvertMode();
+
+converter.addInputs(inputs)
+         .setNewline(Newline.UNIX)
+         .convertToFileInFormat("/tmp/out.xml", "specobject");
+```
+
+#### Using the Tracer From Java
+
+The example below shows how to read documents for the `doc` directory ignoring the artifact types `feat` and `dsn`. The collected specification items are then traced and finally output in a plain text report to STDOUT with verbosity level `ALL`. 
+
+```JAVA
+import org.itsallcode.openfasttrace.Reporter;
+import org.itsallcode.openfasttrace.core.Trace;
+import org.itsallcode.openfasttrace.mode.ReportMode;
+
+final List<String> inputs = Arrays.asList("doc");
+final List<String> ignoreTypes = Arras.asList("feat", "dsn"); 
+final Reporter reporter = new ReportMode();
+
+reporter.addInputs(inputs)
+        .ignoreArtifactTypes(ignoreTypes);
+
+final Trace trace = reporter.trace();
+
+reporter.setReportVerbosity(ReportVerbosity.ALL)
+        .reportToStdOutInFormat(trace, "plain");
+
+if(trace.isAllCovered())
+{
+    // do something
+}
+```
+
+## Tool Support
+
+### Tools for Authoring OFT Documents
+
+The following editors and integrated development environments are well suited for authoring OFT documents. The list is not exhaustive, any editor with Markdown capabilities can be used.
+
+Editor / IDE                                             | Syntax highl. | Preview | Outline | HTML export
+---------------------------------------------------------|---------------|---------|---------|------------
+[Gedit](https://wiki.gnome.org/Apps/Gedit)               | y             |         |         |
+[Eclipse](https://eclipse.org) with WikiText plug-in     | y             | y       | y       | y
+[Eclipse](https://eclipse.org) with GMF plug-in          |               | y       |         |
```

Comment:
Add links to Eclipse editor plugins

Add vscode (https://code.visualstudio.com/): syntax highl=y, preview=y, outline=y